File: src/main/java/com/inti/entities/MissionChauffeur.java
package com.inti.entities;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToMany;
import javax.persistence.ManyToOne;

@Entity
public class MissionChauffeur {
	
	private int idMissionChauffeur;
	private Date dateMission;
	private Date horaireMission;
	
	private Administrateur administrateur;
	
	public MissionChauffeur() {
		super();
	}

	public MissionChauffeur(Date dateMission, Date horaireMission) {
		super();
		this.dateMission = dateMission;
		this.horaireMission = horaireMission;
	}
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	public int getIdMissionChauffeur() {
		return idMissionChauffeur;
	}

	public void setIdMissionChauffeur(int idMissionChauffeur) {
		this.idMissionChauffeur = idMissionChauffeur;
	}

	@Column
	public Date getDateMission() {
		return dateMission;
	}

	public void setDateMission(Date dateMission) {
		this.dateMission = dateMission;
	}

	@Column
	public Date getHoraireMission() {
		return horaireMission;
	}

	public void setHoraireMission(Date horaireMission) {
		this.horaireMission = horaireMission;
	}

	@ManyToOne
	@JoinColumn(name = "AdminResponsableAjout")
	public Administrateur getAdministrateur() {
		return administrateur;
	}

	public void setAdministrateur(Administrateur administrateur) {
		this.administrateur = administrateur;
	}
	
	@ManyToMany(mappedBy = "missionChauffeur")
	private List<MissionChauffeur> listMissionChauffeur = new ArrayList<MissionChauffeur> ();

}

File: src/models/missionChauffeur.ts
import { Time } from '@angular/common';
import { Administrateur } from './administrateur';
import { MoyenTransport } from './moyenTransport';
import { Chauffeur } from './chauffeur';
import { LigneNantes } from './ligneNantes.enum';

export class MissionChauffeur {

    public idMissionChauffeur: number;
    public dateMission: Date;
    public horaireMission: Time;
    public ligneNantes: LigneNantes;

    public administrateur: Administrateur;
    public moyenTransport: MoyenTransport;
    public chauffeur: Chauffeur;

    constructor(dateMission: Date, horaireMission: Time, ligneNantes: LigneNantes, moyenTransport: MoyenTransport,
                chauffeur: Chauffeur) {
        this.dateMission = dateMission;
        this.horaireMission = horaireMission;
        this.ligneNantes = ligneNantes;
        this.moyenTransport = moyenTransport;
        this.chauffeur = chauffeur;
    }

}
